### DISCUSSION TITLE
Is there an Option to start counting Chapter Marks at zero/0?

### DISCUSSION THREAD

**Discussion by Linkinsoldier** [audiamus/AaxAudioConverter#199] [Q&A] [answered]
Hi,

i love the app (Thank you for it!). I do small Tag Editing with Mediamonkey afterwards (my default tool for audio sync and tag manipulation). I could do some of it with AaxAudioConverter, but i'd like to put in in my library afterwards anyway so no harm is done.

There is only think that kind of is a bit bothersome to me. The Opening Credits "Track" always gets the "1" Track/Song Number/Mark and therefore Chapter 1 is Track 2, Chapter 2 is Track 3, ... which kind of is messy (in my opinion) and especially when trying to remember what chapter you listened to last (when switching from music to audiobook). 

This is no big issue, but i am nevertheless asking if there is a way to assign the "Opening Credits" Track the Track/Song Number 0 instead of 1 and then increment as before? If there is (i didnt find it), Chapter 1 would be track1, Chapter 2 would be Track 2 and so on and i would prefer this a lot :)

Thank you,
Linkin

**Comment by audiamus** [comment, marked as answer]
Audible does not use a unique chapter numbering scheme, so this might work for some books but for others. To make things complicated, longer text chapters might be split in two or three by Audible which will confuse the numbering even more.
 
What you can do: If you use BookLibConnect to download, you will usually get the content_meta_<ASIN>.json files. You could edit this file for a particular book. Make sure "use named chapters" is enabled in AaxAudioConverter. 

Better still of course are books that have actual chapter titles in the content_meta file, but they only started this three or four years ago.

**Comment by Linkinsoldier** [comment]
Thanks for the fast response!

I am aware this might not work for every book (i would even say there might be some tweaking necessary for a lot of them), but it would be nice to have the option. 

I am asking for a book with "Chapter x: Chapter Title" and i have the content_meta_.json . You mean removing the opening title from the "chapters" array, right? This would correct the chapter marks (Chapter 1 = Track 1, ... ) but lose me the first title and i have to "convert it twice", right? 

Just to be sure: It did work for me, but I think this is a lot more complicated then an option to "shift the Song#" - or am I missing something?

**Reply by audiamus** [reply]
We have two sets of chapter marks, one embedded as meta data in the audio file, the other in the separate content_meta.json file. For many books, predominantly older publications, both contain only generic chapter names, a simple numbering starting with 1.  For a multipart book, each part starts with 1 again. 

Newer books may have explicit chapter titles instead of just generic chapter numbers. These explicit titles were first found in content_meta.json only but more recently have also appeared in the embedded chapter meta data of the audio file itself. 

If content_meta.json is present and "named chapters" are enabled in AaxAudioConverter, content_meta.json takes precedence to create new embedded chapter meta and also to split by chapter or chapter plus track, if so desired. Unfortunately, content_meta.json chapter timing marks are not always accurate. AaxAudioConverter can apply the silence detection algorithm and some heuristics to adjust chapter marks, but it will not drop chapters. One exception: Audible brand name intros, outros and multipart continuation hints can be removed as an option,

If you manually edit content_meta.json I suggest not to remove any chapter entries, because in that case you will also have to manually update the timings. The built-in heuristics will not be able to fix this.

content_meta.json may also come with a hierarchical chapter structure when downloaded by BookLibConnect from the Audible server, typically for books with multiple logical parts (logical and physical parts are independent of each other). BookLibConnect flattens this hierarchy on export which may result in chapters of zero length. That's identical to what the legacy Audible Win10 app did. AaxAudioConverter will produce non-zero pseudo chapters here, as some audio players cannot deal with zero length.

**Reply by Linkinsoldier** [reply]
thank you for the response. Well. maybe I was just lucky then ;)
At least for my audiobook it worked quite right - I just checked all the converted files and the start and end correctly and are not offseted by the 24s i deleted at the beginning.

We will see if i do it the same way for the next audiobook or manually decrement about 40 numbers ;)

Thank you for the great tooling! Helps me a lot!

**Comment by Linkinsoldier** [comment]
I bought a second Audiobook and it worked as before. I would still argue that this works good, since the start_offset_ms seems to be correct here and it works perfectly for me (i manually checked this here too).

Hope this helps here:
![grafik](https://github.com/audiamus/AaxAudioConverter/assets/24806068/7a467a49-30e3-47da-a849-911f638aed34)
